sgt-back-end/index.js

```diff
@@ -0,0 +1,152 @@
+    .catch(() => {
+      // console.log(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const { name, course } = req.body;
+  const score = Number(req.body.score);
+  // console.log(typeof req.body.score);
+
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    res.status(400).json({
+      error: 'score must be an integer between 0 and 100'
+    });
+    return;
+  }
+
+  if (!name || !course) {
+    res.status(400).json({
+      error: 'name, course, and score are required fields'
+    });
+    return;
+  }
+
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values($1, $2, $3)
+    returning *;
+  `;
+
+  const params = [name, course, score];
+
+  db.query(sql, params)
+    .then(result => {
+      // console.log(result);
+      const [newGrade] = result.rows;
+      res.status(201).json(newGrade);
+    })
+    // .catch(err => {
+    .catch(() => {
+      // console.log(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const id = Number(req.params.gradeId);
+  if (!Number.isInteger(id) || id < 1) {
+    res.status(400).json({
+      error: 'grade must be a positive integer'
+    });
+  }
+  const { name, course } = req.body;
+  const score = Number(req.body.score);
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    res.status(400).json({
+      error: 'score must be an integer between 0 and 100'
+    });
+  }
+  if (!name || !course) {
+    res.status(400).json({
+      error: 'name, course, and score are required fields'
+    });
+  }
+  const sql = `
+    update "grades"
+      set "name" = $1,
+          "course = $2,
+          "score" = $3
+    where "gradeId" = $4
+    returning *
+  `;
+
+  const params = [name, course, score, id];
+  db.query(sql, params)
+    .then(result => {
+      const [updatedGrade] = result.rows;
+      if (!updatedGrade) {
+        res.status(404).json({
+          error: `cannot find grade with gradeId ${id}`
+        });
+      } else {
+        res.json(updatedGrade);
+      }
+    });
+  // .catch(err => next(err));
+});
+
+app.delete('/api/grades/:gradeId', (req, res) => {
+  const id = Number(req.params.gradeId);
+
+  if (!Number.isInteger(id) || id < 1) {
+    res.status(400).json({
+      error: 'grade must be a positive integer'
+    });
+    return;
+  }
+
+  const sql = `
+    delete from "grades"
+    where "gradeId" = $1
+    returning *
+  `;
+
+  const params = [id];
+  db.query(sql, params)
+    .then(result => {
+      // console.log('something happened');
+      const [deletedGrade] = result.rows;
+      if (!deletedGrade) {
+        res.status(404).json({
+          error: `cannot find grade with gradeId ${id}`
+        });
+      } else {
+        res.sendStatus(204);
+      }
+    });
+  // .catch(err => next(err));
```

Comment:
These catches should be included in case querying the database fails. Instead of calling `next()` you can:
- `console.error(err)` to print the error for debugging purposes.
- send a `500` response to inform the client that the back end experienced an error